src/sentry/api/serializers/rest_framework/sentry_app.py

```diff
@@ -34,3 +34,20 @@ class SentryAppSerializer(Serializer):
     redirectUrl = serializers.URLField(required=False)
     isAlertable = serializers.BooleanField(required=False)
     overview = serializers.CharField(required=False)
+
+    def validate_events(self, attrs, source):
+        if not attrs.get('scopes'):
+            return attrs
```

Comment:
I think this makes it so you can get around this validation by just not sending `scopes` in your request. This validator will return here and happily save `events`.

Instead of returning, I think we need to set `scopes` to an empty list so that the check below still happens but fails. Although if they already had the right scope from before, it wouldn't be reflected here since `attrs` is just what's passed from the UI. Is there a way to merge the `attrs` from the request and the current state of the object?

Reply:
per our discussion about ^ adding the check in the mediator for updating an app since the serializer itself gets `partial=True` and we don't have access to the original sentry app scopes to check against

Reply:
If empty values imply no update how do you discern an omitted key and intentionally blanking the scopes?

Reply:
Intentionally blank should be an empty array, I think.

We're making up for this logic weirdness in the Mediator by effectively doing the _real_ validation (apply `scope` changes to the object, then validate `events` against the new set of scopes). These Serializer (Validator) objects are strange in that they're only validating the data coming in, not the resulting state of the resource after they're applied. 🤷‍♂️